atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/bucketingthings/DefaultSweepAssignedBucketStore.java

```diff
@@ -174,6 +185,32 @@ public void updateStartingBucketForShardAndStrategy(Bucket newStartingBucket) {
         }
     }
 
+    @Override
+    public Set<SweepableBucket> getSweepableBuckets(Set<Bucket> startBuckets) {
+        List<SweepAssignedBucketsRow> rows = startBuckets.stream()
+                .flatMap(bucket -> Stream.of(
+                        SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketsRow(bucket),
+                        SweepAssignedBucketStoreKeyPersister.INSTANCE.nextSweepBucketsRow(bucket)))
```

Comment:
Might be worth a comment here explaining the logic of choosing two rows.

Reply:
Yeah I have a long comment I'll push up later, it got lost in the exercise of splitting all these PRs up at speed.